src/Microsoft.Health.Dicom.Functions.UnitTests/Management/PurgeOrchestrationInstancesHistoryTests.cs

```diff
@@ -0,0 +1,132 @@
+            _definedNow = DateTime.UtcNow;
+            _purgeOrchestrationInstancesHistory = new PurgeOrchestrationInstancesHistory(
+                Options.Create(_purgeConfig),
+                () => _definedNow);
+            _durableOrchestrationClientMock = Substitute.For<IDurableOrchestrationClient>();
+        }
+
+        [Fact]
+        public async Task GivenValidInput_PurgeCompletedDurableFunctionsHistory_ShouldNotPurgeAnythingOrchestrationsAsync()
+        {
+            using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+
+            _durableOrchestrationClientMock
+                .ListInstancesAsync(
+                    Arg.Is<OrchestrationStatusQueryCondition>(givenCondition => IsCondition(givenCondition, null)),
+                    Arg.Is<CancellationToken>(cancellationTokenSource.Token))
+                .Returns(Task.FromResult(new OrchestrationStatusQueryResult { DurableOrchestrationState = Array.Empty<DurableOrchestrationStatus>() }));
+
+            await _purgeOrchestrationInstancesHistory.Run(_timer, _durableOrchestrationClientMock, _logger, cancellationTokenSource.Token);
+
+            await _durableOrchestrationClientMock
+                .DidNotReceiveWithAnyArgs()
+                .PurgeInstanceHistoryAsync(Arg.Any<string>());
+        }
+
+        [Fact]
+        public async Task GivenValidInput_PurgeCompletedDurableFunctionsHistory_ShouldPurgeMultiPageListOfOrchestrationsAsync()
+        {
+            using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+
+            List<DurableOrchestrationStatus> durableOrchestrationStatuses = new List<DurableOrchestrationStatus>()
+            {
+                new DurableOrchestrationStatus{ InstanceId = "1" },
+                new DurableOrchestrationStatus{ InstanceId = "2" },
+                new DurableOrchestrationStatus{ InstanceId = "3" }
+            };
+
+            OrchestrationStatusQueryResult orchestrationStatusQueryResult = new OrchestrationStatusQueryResult();
+            orchestrationStatusQueryResult.DurableOrchestrationState = durableOrchestrationStatuses;
+            orchestrationStatusQueryResult.ContinuationToken = "fake_cancellation_token";
+
+            var collector = new List<string>();
+
+            _durableOrchestrationClientMock
+                .ListInstancesAsync(
+                    Arg.Is<OrchestrationStatusQueryCondition>(
+                            givenCondition => IsCondition(givenCondition, null)),
+                    Arg.Is<CancellationToken>(cancellationTokenSource.Token))
+                // First page, should have 3 orchestrations, then change continuation token.
+                .Returns(_ =>
+                    {
+                        orchestrationStatusQueryResult.ContinuationToken = null;
+                        Assert.True(orchestrationStatusQueryResult.DurableOrchestrationState.SequenceEqual(durableOrchestrationStatuses));
```

Comment:
This isn't an appropriate place to `Assert` behavior as you're configuring it here. Instead, you should return a new `orchestrationStatusQueryResult` object with the desired state